### PR TITLE
Feature/compare all targets search params

### PR DESCRIPTION
**Pull request by kfilip94**
This PR saves the document selection in the URL as a `targets` param on `Compare all targets`  and  `Custom compare` pages. Navigation through `Go to compare all targets` button passes the `targets` params to the `Compare all targets` page. It also removes the `Custom compare` option from the `Commitments` menu.
[PIVOTAL](https://www.pivotaltracker.com/story/show/171315679)

mc1l1-38xzt

I hope that this solution is not overcomplicated, let me know what do you think ;)

### CODE REVIEW

**Comment by kfilip94** [comment]
@Bluesmile82 ready for reviewing!

File: app/javascript/app/pages/custom-compare/custom-compare-selectors.js [hunk 1]
```diff
@@ -17,42 +27,39 @@ const getCountryOptions = createSelector([getCountries], countries => {
...
-    document0,
-    document1,
-    document2
-  };
+  const queryTargets = query.targets ? query.targets.split(',') : [];
+  return [1, 2, 3].map((value, i) => {
+    // targets are saved as a string 'ISO3-DOCUMENT', e.g. 'USA-NDC'
+    const target =
+      queryTargets && queryTargets[i] && queryTargets[i].split('-');
+    const country = target && target.length > 0 && target[0];
```
Comment (Bluesmile82) [inline app/javascript/app/pages/custom-compare/custom-compare-selectors.js]: Why this extra length > 0? And shouldn't it be only length > 0 for document?

File: app/javascript/app/pages/custom-compare/custom-compare.js [hunk 2]
```diff
@@ -17,19 +22,52 @@ const mapStateToProps = (state, { location, route }) => {
...
+    };
     history.replace(getLocationParamUpdated(location, params));
   };
 
-  return <Component {...props} handleFilterChange={handleFilterChange} />;
+  const handleCountryFilterChange = (targetKey, newCountry) => {
+    const { selectedTargets } = props;
+    const newTargetParams = [];
+    selectedTargets.forEach(({ key, country, document = '' }) => {
+      if (key === targetKey) newTargetParams.push(`${newCountry}-`);
```
Comment (Bluesmile82) [inline app/javascript/app/pages/custom-compare/custom-compare.js]: Shouldn't we always have a country and a document? Why add a country without a document?

Comment (kfilip94) [reply app/javascript/app/pages/custom-compare/custom-compare.js]: After changing the country we're not selecting automatically the document, we leave the dropdown empty. The question is what do we want to do with that @fannycc @Bluesmile82?
vaq2b-z5etk

File: app/javascript/app/pages/custom-compare/custom-compare-selectors.js (hunk 1 above)
Comment (kfilip94) [reply app/javascript/app/pages/custom-compare/custom-compare-selectors.js]: :man_facepalming: Oh, you're right,  we don't need to check `target.length` in both cases, we will get `undefined` anyway, thanks!

**Comment by kfilip94** [comment]
Hey @Bluesmile82, ready again!

**Comment by Bluesmile82** [comment]
Great! I can see now that we have the case of Lybia without documents. Works perfectly!

**Pull request merged**